bindings/sundials4py/cvodes/cvodes_generated.hpp

```diff
@@ -0,0 +1,2035 @@
+      nb::arg("which"), nb::arg("constraintsB"));
+
+m.def("CVodeSetQuadErrConB", CVodeSetQuadErrConB, nb::arg("cvode_mem"),
+      nb::arg("which"), nb::arg("errconQB"));
+
+m.def("CVodeSetNonlinearSolverB", CVodeSetNonlinearSolverB,
+      nb::arg("cvode_mem"), nb::arg("which"), nb::arg("NLS"));
+
+m.def(
+  "CVodeGetB",
+  [](void* cvode_mem, int which, N_Vector yB) -> std::tuple<int, sunrealtype>
+  {
+    auto CVodeGetB_adapt_modifiable_immutable_to_return =
+      [](void* cvode_mem, int which, N_Vector yB) -> std::tuple<int, sunrealtype>
+    {
+      sunrealtype tBret_adapt_modifiable;
+
+      int r = CVodeGetB(cvode_mem, which, &tBret_adapt_modifiable, yB);
+      return std::make_tuple(r, tBret_adapt_modifiable);
+    };
+
+    return CVodeGetB_adapt_modifiable_immutable_to_return(cvode_mem, which, yB);
+  },
+  nb::arg("cvode_mem"), nb::arg("which"), nb::arg("yB"));
+
+m.def(
+  "CVodeGetQuadB",
+  [](void* cvode_mem, int which, N_Vector qB) -> std::tuple<int, sunrealtype>
+  {
+    auto CVodeGetQuadB_adapt_modifiable_immutable_to_return =
+      [](void* cvode_mem, int which, N_Vector qB) -> std::tuple<int, sunrealtype>
+    {
+      sunrealtype tBret_adapt_modifiable;
+
+      int r = CVodeGetQuadB(cvode_mem, which, &tBret_adapt_modifiable, qB);
+      return std::make_tuple(r, tBret_adapt_modifiable);
+    };
+
+    return CVodeGetQuadB_adapt_modifiable_immutable_to_return(cvode_mem, which,
+                                                              qB);
+  },
+  nb::arg("cvode_mem"), nb::arg("which"), nb::arg("qB"));
+
+m.def("CVodeGetAdjCVodeBmem", CVodeGetAdjCVodeBmem, nb::arg("cvode_mem"),
+      nb::arg("which"));
+
+m.def("CVodeGetAdjY", CVodeGetAdjY, nb::arg("cvode_mem"), nb::arg("t"),
+      nb::arg("y"));
+
+m.def("CVodeGetAdjCheckPointsInfo", CVodeGetAdjCheckPointsInfo,
+      nb::arg("cvode_mem"), nb::arg("ckpnt"));
+
+m.def(
+  "CVodeGetAdjDataPointHermite",
+  [](void* cvode_mem, int which, N_Vector y,
+     N_Vector yd) -> std::tuple<int, sunrealtype>
+  {
+    auto CVodeGetAdjDataPointHermite_adapt_modifiable_immutable_to_return =
+      [](void* cvode_mem, int which, N_Vector y,
+         N_Vector yd) -> std::tuple<int, sunrealtype>
+    {
+      sunrealtype t_adapt_modifiable;
+
+      int r = CVodeGetAdjDataPointHermite(cvode_mem, which, &t_adapt_modifiable,
+                                          y, yd);
+      return std::make_tuple(r, t_adapt_modifiable);
+    };
+
+    return CVodeGetAdjDataPointHermite_adapt_modifiable_immutable_to_return(cvode_mem,
+                                                                            which,
+                                                                            y,
+                                                                            yd);
+  },
+  nb::arg("cvode_mem"), nb::arg("which"), nb::arg("y"), nb::arg("yd"));
+
+m.def(
+  "CVodeGetAdjDataPointPolynomial",
+  [](void* cvode_mem, int which, N_Vector y) -> std::tuple<int, sunrealtype, int>
```

Comment:
`y` is allowed to be `NULL`, does this need special handling like the `A` matrix in `CVodeSetLinearSolver`?